src/Ocelot/Requester/HttpExeptionToErrorMapper.cs

```diff
@@ -28,6 +30,11 @@ public Error Map(Exception exception)
                 return new RequestCanceledError(exception.Message);
             }
 
+            if (type == typeof(HttpRequestException))
+            {
+                return new ConnectionToDownstreamServiceError(exception);
+            }
```

Comment:
Maybe it would be good to check if the inner exception is of type **SocketException** and the **SocketErrorCode** is **HostNotFound**.

Reply:
I considered checking the inner exception and ConnectionRefused error code (the error code from the issue #603), but when I analyzed this issue more, I decided to generalize these conditions to catch other situations where there is no connection to the downstream service.

See: https://github.com/ThreeMammals/Ocelot/issues/603

Reply:
Ok, sounds reasonable.